src/main/kotlin/no/nav/syfo/client/kodeverk/KodeverkClient.kt

```diff
@@ -0,0 +1,98 @@
+package no.nav.syfo.client.kodeverk
+
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.coroutines.channels.ClosedReceiveChannelException
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.application.cache.RedisStore
+import no.nav.syfo.client.httpClientDefault
+import no.nav.syfo.util.*
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+
+class KodeverkClient(
+    private val redisStore: RedisStore,
+    private val baseUrl: String,
+) {
+    private val httpClient = httpClientDefault()
+
+    suspend fun getPostinformasjon(
+        callId: String,
+    ): List<Postinformasjon> {
+        val cachedValue = redisStore.getListObject<Postinformasjon>(CACHE_POSTINFORMASJON_KEY)
+
+        if (cachedValue != null) {
+            return cachedValue
+        } else {
+            val postinformasjon = getPostnummerKodeverk(
+                callId = callId,
+            )
+            if (postinformasjon.isNotEmpty()) {
+                redisStore.setObject(
+                    expireSeconds = CACHE_KODEVERK_POSTINFORMASJON_EXPIRE_SECONDS,
+                    key = CACHE_POSTINFORMASJON_KEY,
+                    value = postinformasjon,
+                )
+            }
+
+            return postinformasjon
+        }
+    }
+
+    private suspend fun getPostnummerKodeverk(
+        callId: String,
+    ): List<Postinformasjon> {
+        val postnummerUrl = "$baseUrl$KODEVERK_POSTNUMMER_BETYDNINGER_PATH"
+        return try {
+            val response: HttpResponse = httpClient.get(postnummerUrl) {
```

Comment:
Trengs ingen token for denne tjenesten forresten?

Reply:
Nei, den er åpen 👍